app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -134,6 +148,19 @@ private class HomeTileAdapter(val onUrlEnteredListener: OnUrlEnteredListener, ho
 
     override fun getItemCount() = tiles.size
 
+    fun deleteTile(id: String) {
+        var blacklist = context.getSharedPreferences(HOME_TILES_PREFS, MODE_PRIVATE).getStringSet(BLACKLIST, null)
+        if (blacklist != null) {
+            blacklist.add(id)
+        } else {
+            blacklist = mutableSetOf(id)
+        }
+        val edit = context.getSharedPreferences(HOME_TILES_PREFS, MODE_PRIVATE).edit()
+        edit.clear() // maybe copy array??
```

Comment:
Yeah, fix this - I think the fact that this works is an unintended side effect (and it's not mentioned in the docs as a way to fix the "don't modify the string set" problem), so we shouldn't rely on it.